test/e2e/util/resource_queue.go

```diff
@@ -0,0 +1,100 @@
+package util
+
+import (
+	"fmt"
+	k8scontrollerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sync"
+)
+
+type ResourceQueue struct {
+	queue       []k8scontrollerclient.Object
+	lookupTable map[k8scontrollerclient.Object]bool
```

Comment:
In Go, the convention for a set is `map[T]struct`; the boolean value is unnecessary.